packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2747,4 +2957,301 @@ export class TransactionController extends BaseControllerV1<
       log('Error while updating post transaction balance', error);
     }
   }
+
+  #getEthQuery({
+    networkClientId,
+    chainId,
+  }: {
+    networkClientId?: NetworkClientId;
+    chainId?: Hex;
+  }): EthQuery {
+    // if multichain is disabled, use the global ethQuery
+    if (!this.#enableMultichain) {
+      return this.ethQuery;
+    }
+    let networkClient: NetworkClient | undefined;
+
+    if (networkClientId) {
+      try {
+        networkClient = this.messagingSystem.call(
+          `NetworkController:getNetworkClientById`,
+          networkClientId,
+        );
+      } catch (err) {
+        log('failed to get network client by networkClientId');
+      }
+    }
+
+    if (!networkClient && chainId) {
+      try {
+        networkClientId = this.messagingSystem.call(
+          `NetworkController:findNetworkClientIdByChainId`,
+          chainId,
+        );
+        networkClient = this.messagingSystem.call(
+          `NetworkController:getNetworkClientById`,
+          networkClientId,
+        );
+      } catch (err) {
+        log('failed to get network client by chainId');
+      }
+    }
+
+    if (networkClient) {
+      return new EthQuery(networkClient.provider);
+    }
+
+    // NOTE(JL): we're not ready to drop globally selected ethQuery yet.
+    // Some calls to getEthQuery only have access to optional networkClientId
+    // throw new Error('failed to get eth query instance');
+    return this.ethQuery;
+  }
+
+  #refreshEtherscanRemoteTransactionSources = (
+    networkClients: NetworkClientRegistry,
+  ) => {
+    // this will be prettier when we have consolidated network clients with a single chainId:
```

Comment:
Out of curiosity, putting aside how things work now, what would you prefer to be able to write here? What does "prettier" look like?

Reply:
I think the idea is that
```
    const chainIdsInRegistry = new Set();
    Object.values(networkClients).forEach((networkClient) =>
      chainIdsInRegistry.add(networkClient.configuration.chainId),
    );
```
becomes
```
  const chainIdsInRegistry = Object.values(networkClients).map((networkClient) => networkClient.configuration.chainId)
```

because there should only be one network client active per chain ID